### PR TITLE
added Fire Linux OS (FirePOWER) and SonicOS (Sonicwall) support

### PR DESCRIPTION
**Pull request by rgnv**

### CODE REVIEW

File: lib/oxidized/model/aireos.rb [hunk 1]
```diff
@@ -45,6 +45,7 @@ def clean cfg
     cfg.each_line do |line|
       next if line.match /^\s*$/
       next if line.match /rogue (adhoc|client) (alert|Unknown) [\da-f]{2}:/
+      next if line.match /interface nat-address management set \d+.\d+.\d+.\d+/
```
Comment (danilopopeye) [inline lib/oxidized/model/aireos.rb]: not sure this belong to this PR. this was intended?

File: lib/oxidized/model/sonicos.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    cfg.gsub! /^: [^\n]*\n/, ''
+    clean cfg
+  end
+
+  cfg :telnet, :ssh do
+    post_login 'no cli pager session'
+    pre_logout 'exit'
+  end
+
+  def clean cfg
```
Comment (danilopopeye) [inline lib/oxidized/model/sonicos.rb]:
```ruby
def clean cfg
  cfg.map do |line|
    next if line.match /system-time \"\d{2}\/\d{2}\/\d{4} \d{2}\:\d{2}:\d{2}.\d+\"/
    next if line.match /system-uptime "(\s+up\s+\d+\s+)|(.*Days.*)"/
    next if line.match /checksum \d+/
    line.strip
  end.compact.join("\n") << "\n"
end
```

wdyt? 😃

File: lib/oxidized/model/sonicos.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    out = []
+    cfg.each_line do |line|
+      next if line.match /system-time \"\d{2}\/\d{2}\/\d{4} \d{2}\:\d{2}:\d{2}.\d+\"/
+      next if line.match /system-uptime "(\s+up\s+\d+\s+)|(.*Days.*)"/
+      next if line.match /checksum \d+/
+      line = line[1..-1] if line[0] == "\r"
```
Comment (danilopopeye) [inline lib/oxidized/model/sonicos.rb]: the `#strip` in the next line would remove the `\r`, so we can remove this line.

```ruby
"\tgoodbye\r\n".strip   #=> "goodbye"
```

**Comment by rgnv** [comment]
Wrong branch pushed out for pull request.

**Pull request closed without merging**